RELEASE.md

```diff
@@ -9,17 +9,32 @@ The Service APIs project is an API project that has the following two components
 This repository is the home for both of the above components.
 
 ## Versioning strategy
+<!-- Content here should be mirrored in docs-src/releases.md -->
+This project uses 2 different but related forms of versioning:
 
-Releases in this repository follow Go module versioning conventions and
-semantic versioning.
-During the `alpha` and `beta` stage of the API, version tags will take the form
-of `v0.x.y`.
-Minor version must be incremented whenever a new API version is introduced for
-any resource or even when smaller backwards-compatible additions are made to the API.
-Bug fixes and clarifications in the spec will lead to patch number being incremented.
+- [Kubernetes API Versioning] (example: v1alpha1)
+- [Semantic Versioning] (example: v0.1.0)
 
-During GA (when `apiVersion` changes to `v1`), the Git version tag will be bumped
-up to `v1.0.0`.
+Each new API version will be released with a new semantic version. For example,
+v1alpha1 was released with v0.1.0. Before we release an API version, we may
+provide some release candidates such as v0.1.0-rc1 as a way to evaluate a new
+API version before it is formally released. All releases will be compatible with
+[Go modules versioning].
+
+As the API evolves, we will make intermediate releases that improve upon an
+existing API version. These releases will be fully backwards compatible and will
+be limited to either bug fixes or additions to the API.
```

Comment:
Should we clarify how releases for new API versions and incremental updates relate to minor and patch numbers?

Reply:
Good call, added something here but it still feels a bit clunky - open to any ideas on how to improve it.

Reply:
It is a little clunky but it gets the point across on what to expect in future. Resolving this thread. Thanks!